### DISCUSSION TITLE
2011-11-15 Core team update

### DISCUSSION THREAD

**Discussion by mm-near** [near/community#126] [Core]
# Highlights:
* sharding arriving in Mainnet this week (Phase 0)

# TL;DR
@pmnoxx - improving network performance / network rate-limiting
@Longarithm - improving fees / debugging performance inconsistencies
@mzhangmzz -  sharding mainnet launch / improve chunk fetching
@yordanmadzhunkov - improve fast-testing (have deterministic tests) before our 'more validators' launch on Nov 29
@EgorKulikov - working on different validator assignment (for 'more validators' on Nov 29)
@abacabadabacaba - better storage analysis (replace Merkle tree with AVL)

# Done:
@pmnoxx - lots of network refactoring code (around 20 PRs) with documentation improvements
@pmnoxx - Contritube to DeepSize repo (Aeledfyr/deepsize#19)
@pmnoxx - Fixes for Betanet incident 2021-11-08 (near/nearcore#5219)
@pmnoxx - Shipped first part of rate limiter changes (https://github.com/near/nearcore/pull/49270
@pmnoxx - Shipped moving routing table to another thread (near/nearcore#5089)
@pmnoxx - Extra logs to debug why nodes are failing out of sync (near/nearcore#5186)

@Longarithm - estimating cost of calling no-op functions - Findings: switching from rkyv serializing to archiving can reduce time 2x; for all known cases, maximal current underestimation is <4x.
@Longarithm - Investigating contract where out estimation was 10x off - seems related to near/nearcore#4771
@Longarithm - code cleanup: removing default impl of VMConfig (near/nearcore#5204)

@mzhangmzz - Landed the fix for ViewClient::GetExecutionOutcome near/nearcore#5153
@mzhangmzz - Found a bug in the implementation of storing PartialEncodedChunk where we only store one for each chunk, and submitted a fix for that near/nearcore#5201
@mzhangmzz - Submit a PR to implement requesting chunks for multiple blocks simultaneously near/nearcore#5224

# Progress:
@pmnoxx - Improve routing table performance 5x (near/nearcore#5221)
@pmnoxx - Support for DeepSize for all codebase (Aeledfyr/deepsize#18)
@pmnoxx - NearClock - a wrapper around SystemClock, which is compatible with existing Clock mocking (near/nearcore#5177)

@Longarithm - working on separating config fields doc

@mzhangmzz - oncall for mainnet upgrade

@EgorKulikov - Worked on splitting new validator selection algorithm from chunk only producer feature
@EgorKulikov - work on NEP for in memory contract bidding
